sw/device/silicon_creator/lib/base/mock_rom_mmio.h

```diff
@@ -0,0 +1,114 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BASE_MOCK_ROM_MMIO_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BASE_MOCK_ROM_MMIO_H_
+
+#include "sw/device/lib/base/testing/mock_mmio_test_utils.h"
+#include "sw/device/lib/testing/mask_rom_test.h"
+#include "sw/device/silicon_creator/lib/base/rom_mmio.h"
+
+namespace mask_rom_test {
+namespace internal {
+/**
+ * Mock class for rom_mmio.c.
+ */
+class MockRomMmio {
+ public:
+  MOCK_METHOD(uint32_t, Read8, (uint32_t addr));
+  MOCK_METHOD(void, Write8, (uint32_t addr, uint32_t value));
+  MOCK_METHOD(uint32_t, Read32, (uint32_t addr));
+  MOCK_METHOD(void, Write32, (uint32_t addr, uint32_t value));
+
+  virtual ~MockRomMmio() {}
+
+ protected:
+  ::testing::InSequence seq_;
+};
+}  // namespace internal
+
+using MockRomMmio = GlobalMock<testing::StrictMock<internal::MockRomMmio>>;
+
+/**
+ * Expect a read to the device `dev` at the given offset, returning the given
+ * 8-bit value.
+ *
+ * The value may be given as an integer, a pointer to little-endian data,
+ * or a `std::initializer_list<BitField>`.
+ *
+ * This expectation is sequenced with all other `EXPECT_READ` and `EXPECT_WRITE`
+ * calls.
+ */
+#define EXPECT_MMIO_READ8(mmio, addr, ...) \
```

Comment:
I feel like `EXPECT_MMIO_READ` v `EXPECT_READ` will be a bit confusing...

Reply:
Should I call it just `EXPECT_{READ,WRITE}` or other prefix?

Reply:
Updated to `EXPECT_{READ,WRITE}`

Reply:
Actually, I would have gone with `EXPECT_ABS_READ`... I'm not going to pick on this more than that though.

Reply:
Updated with @mcy's recommended names.